src/canisters/frontend/ic-asset/src/batch_upload/operations.rs

```diff
@@ -167,3 +175,54 @@ pub(crate) fn set_encodings(
         }
     }
 }
+
+pub(crate) fn update_properties(
+    operations: &mut Vec<BatchOperationKind>,
+    project_assets: &HashMap<String, ProjectAsset>,
+    canister_asset_properties: &HashMap<String, AssetProperties>,
+) {
+    for (key, project_asset) in project_assets {
+        let project_asset_properties = project_asset.asset_descriptor.config.clone();
+        let canister_asset_properties = canister_asset_properties.get(key);
+        // skip if the asset is not already in the canister, because
+        // proporties gonna be created during create_new_assets call
+        if canister_asset_properties.is_none() {
+            continue;
+        }
+        let canister_asset_properties = canister_asset_properties.unwrap();
+        let cache_is_different = project_asset_properties
+            .cache
+            .as_ref()
+            .and_then(|v| v.max_age)
+            != canister_asset_properties.max_age;
+        let headers_are_different = project_asset_properties.headers
+            != canister_asset_properties
+                .headers
+                .as_ref()
+                .map(|v| BTreeMap::from_iter(v.clone().into_iter()));
+        let allow_raw_access_is_different =
+            project_asset_properties.allow_raw_access != canister_asset_properties.allow_raw_access;
+
+        // check if the properties are the same and skip if they are to save saves cycles
+        if cache_is_different || headers_are_different || allow_raw_access_is_different {
+            operations.push(BatchOperationKind::SetAssetProperties(
+                SetAssetPropertiesArguments {
+                    key: key.clone(),
+                    max_age: Some(
+                        project_asset_properties
+                            .cache
+                            .as_ref()
+                            .and_then(|c| c.max_age),
+                    ),
+                    headers: Some(project_asset_properties.headers.map(|hm| {
+                        hm.iter()
+                            .map(|(k, v)| (k.clone(), v.clone()))
+                            .collect::<Vec<_>>()
+                    })),
+                    is_aliased: Some(project_asset_properties.enable_aliasing),
+                    allow_raw_access: Some(project_asset_properties.allow_raw_access),
+                },
+            ));
+        }
+    }
```

Comment:
Let's set these individually.  They are all `opt opt` in order to support this.  Meaning all these values start as None, and then:
- set max_age to Some(new max age) if max age is different
- set headers to Some(new headers) if headers are different
- set is_aliased to Some(new value) if enable_aliasing is different
- set allow_raw_access to Some(new value) if allow_raw_access is different

This is a little more efficient, but more importantly, notice that the check on line 207 didn't check for `enable_aliasing` to change, so if it was the only thing that changed, no SetAssetProperties operation would be pushed.

Please start by updating a test, because this code should have failed at least four assertions.